### PR TITLE
Add setting and expose prometheus on port 9100

### PR DESCRIPTION
**Pull request by rodrigok**
# Instrumentations:
- [x] methods
- [x] pub/sub
- [x] callbacks
- [x] statistics
- [x] REST API
- [x] Notifications
- [ ] Avatar requests
- [ ] Upload requests
- [ ] User session time

### CODE REVIEW

**Comment by sampaiodiego** [comment]
we coud also add:

- [ ] methods response time
- [ ] pub/sub response time

File: packages/rocketchat-lib/server/lib/metrics.js [hunk 1]
```diff
@@ -1,9 +1,138 @@
...
+});
+
+const server = http.createServer(app);
+
+let timer;
+RocketChat.settings.get('Prometheus_Enabled', (key, value) => {
+	if (value === true) {
+		server.listen({
+			port: 9100,
+			host: process.env.BIND_IP || '0.0.0.0'
```
Comment (geekgonecrazy) [inline packages/rocketchat-lib/server/lib/metrics.js]: Can we re-use instance ip?  I don't think we are using bind ip anywhere

Comment (rodrigok) [reply packages/rocketchat-lib/server/lib/metrics.js]: @geekgonecrazy that is a copy from internal meteor code

File: packages/rocketchat-lib/server/lib/debug.js [hunk 2]
```diff
@@ -35,6 +38,7 @@ const originalMeteorPublish = Meteor.publish;
 Meteor.publish = function(name, func) {
 	return originalMeteorPublish(name, function() {
 		logger.publish(name, '-> userId:', this.userId, ', arguments: ', arguments);
+		RocketChat.metrics.meteorSubscriptions.inc({subscription: name}, 1, new Date());
```
[the remark this answers is not in the document]
Comment (rodrigok) [reply packages/rocketchat-lib/server/lib/debug.js]: You are right, the response time would be the initial response time?

[the remark this answers is not in the document]
Comment (sampaiodiego) [reply packages/rocketchat-lib/server/lib/debug.js]: it's the initial time until `ready()` being called.

**Comment by floyd871** [comment]
Why is this hardwired on 9100??  This port is normaly used by the node-exporter and do a conflict on our systems. 
Please do it configurable.

**Comment by thaiphv** [comment]
Can we make the port configurable? It conflicts with the `node-exporter`. See https://github.com/prometheus/prometheus/wiki/Default-port-allocations

**Comment by thaiphv** [comment]
I've created a new issue suggesting the feature. See https://github.com/RocketChat/Rocket.Chat/issues/11114

**Pull request merged**